### ISSUE TITLE
Initial `quarkus:dev` logging improvements

### ISSUE THREAD

**Issue by FroMage** [quarkusio/quarkus#17808]
I just got greeted by:

```
2021-06-09 17:13:59,367 WARN  [io.sma.graphql] (Quarkus Main Thread) SRGQL010000: Schema is null, or it has no operations. Not bootstrapping SmallRye GraphQL
2021-06-09 17:13:59,561 INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service
2021-06-09 17:13:59,635 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) grpc-graphql 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.980s. Listening on: http://localhost:8080
2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-server, resteasy-reactive, smallrye-context-propagation, smallrye-graphql, vertx]
```

And I have several remarks:

- Is the GraphQL `WARN` really useful? @phillip-kruger WDYT? If not, we should just remove it
- IMO the `[io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service` is useless and should be dropped, WDYT @michalszynkiewicz ?
- The `[io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]` line kinda has useful info, I guess? But then it seems to me we should merge it with the following line: `Listening on: http://localhost:8080` and perhaps split it into something like:

```
2021-06-09 17:13:59,635 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) grpc-graphql 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.980s. 
2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) Listening on: http://localhost:8080
2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) DEV UI: http://localhost:8080/q
2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) gRPC server: http://0.0.0.0:9000
2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-server, resteasy-reactive, smallrye-context-propagation, smallrye-graphql, vertx]
```

Notice I added a link to the DEV UI which has always been missing IMO.
Side question: is it only me finding it weird that Quarkus runs on `localhost` while gRPC runs on `0.0.0.0` @michalszynkiewicz ?

**Comment by FroMage** [comment]
TBH at this point I also wonder why we even have the `2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) ` prefixes for our initial info dump, since we managed to get our banner to not have this. What purpose does it serve?

**Comment by phillip-kruger** [comment]
@FroMage yea I do not have strong opinions on this. Happy to remove or change the level of the graphql logging

**Comment by michalszynkiewicz** [comment]
śr., 9 cze 2021 o 17:22 Stéphane Épardaud ***@***.***>
napisał(a):

> I just got greeted by:
>
> 2021-06-09 17:13:59,367 WARN  [io.sma.graphql] (Quarkus Main Thread) SRGQL010000: Schema is null, or it has no operations. Not bootstrapping SmallRye GraphQL
> 2021-06-09 17:13:59,561 INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service
> 2021-06-09 17:13:59,635 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
> 2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) grpc-graphql 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.980s. Listening on: http://localhost:8080
> 2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
> 2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-server, resteasy-reactive, smallrye-context-propagation, smallrye-graphql, vertx]
>
> And I have several remarks:
>
>    - Is the GraphQL WARN really useful? @phillip-kruger
>    <https://github.com/phillip-kruger> WDYT? If not, we should just
>    remove it
>    - IMO the [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread)
>    Registering gRPC reflection service is useless and should be dropped,
>    WDYT @michalszynkiewicz <https://github.com/michalszynkiewicz> ?
>    - The [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0)
>    gRPC Server started on 0.0.0.0:9000 [SSL enabled: false] line kinda
>    has useful info, I guess? But then it seems to me we should merge it with
>    the following line: Listening on: http://localhost:8080 and perhaps
>    split it into something like:
>
> 2021-06-09 17:13:59,635 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
> 2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) grpc-graphql 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.980s.
> 2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) Listening on: http://localhost:8080
> 2021-06-09 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) DEV UI: http://localhost:8080/q
> 2021-06-09 <http://localhost:8080/q2021-06-09> 17:13:59,714 INFO  [io.quarkus] (Quarkus Main Thread) gRPC server: http://0.0.0.0:9000
> 2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
> 2021-06-09 17:13:59,715 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-server, resteasy-reactive, smallrye-context-propagation, smallrye-graphql, vertx]
>
> Notice I added a link to the DEV UI which has always been missing IMO.
> Side question: is it only me finding it weird that Quarkus runs on
> localhost while gRPC runs on 0.0.0.0 @michalszynkiewicz
> <https://github.com/michalszynkiewicz> ?
>

I inherited it this way from Clement :D
+1 for aligning it. What's the motivation to expose http on localhost only?
Security?

I also like the idea of having all the messages of this type printed
together, in the example above, I think these two lines should be merged:

[io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server
started on 0.0.0.0:9000 [SSL enabled: false]
[io.quarkus] (Quarkus Main Thread) gRPC server: http://0.0.0.0:9000


IMHO, info wether gRPC reflection service is started or not is valuable,
with it one can use e.g. grpcurl to test the service. But it definitely
doesn't deserve its own log line, maybe all three lines could be compacted
to the following?

[io.quarkus] (Quarkus Main Thread) gRPC server: http://0.0.0.0:9000 [SSL
enabled: false, reflection enabled: true]

**Comment by FroMage** [comment]
Yeah, that'd work. Though not sure why SSL info is needed, since everyone can see it's http and not https?

**Comment by michalszynkiewicz** [comment]
actually, grpc world doesn't seem to use neither `http` nor `https` schema in the URLs, so for gRPC we'd probably want:
```
[io.quarkus] (Quarkus Main Thread) gRPC server: 0.0.0.0:9000 [SSL
enabled: false, reflection enabled: true]
```

**Comment by FroMage** [comment]
I've changed my mind, I think it's useless now to display the DEV UI URL because I can press `d` on the DEV console and get there.

**Comment by mkouba** [comment]
So `INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service` and `INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]` are still there. I did not try the GraphQL WARN.

I think tha the `gRPC Server started ...` is useful and does not need to be merged with `Listening on...` which is already merged with the `grpc-graphql 1.0.0-SNAPSHOT on JVM` line. Not sure about the "reflection service". CC @cescoffier 

> I've changed my mind, I think it's useless now to display the DEV UI URL because I can press d on the DEV console and get there.

+1

@FroMage do you still think the logging should be improved?

**Comment by cescoffier** [comment]
Indicating that the reflection service is useful to know if you can use the gRPC toolings (like grpcurl). 

It's something we enabled by default in dev but disable by default in prod. So, for a regular users it may not be simple to understand if it's enabled or not.

**Comment by mkouba** [comment]
> Indicating that the reflection service is useful to know if you can use the gRPC toolings (like grpcurl).
> 
> It's something we enabled by default in dev but disable by default in prod. So, for a regular users it may not be simple to understand if it's enabled or not.

Thanks @cescoffier.  In that case, gRPC is fine.

**Comment by FroMage** [comment]
OK but a single line as suggested by Michal was a nice idea: `gRPC server: 0.0.0.0:9000 [SSL
enabled: false, reflection enabled: true]`

**Comment by cescoffier** [comment]
Yes, a single statement would be nice. Actually, with the new gRPC server I would change it a bit and only indicate the gRPC port if it's not the http one.
